zqd/ingest/pcap.go

```diff
@@ -210,6 +268,29 @@ func (p *PcapOp) createSnapshot(ctx context.Context) error {
 	return nil
 }
 
+func (p *PcapOp) convertSuricataLog(ctx context.Context) error {
+	// For now, this just converts from json to
+	// zng. Soon it will apply a typing config and rename at least
+	// the timestamp field.
+	zctx := resolver.NewContext()
+	path := filepath.Join(p.logdir, "eve.json")
+	zr, err := detector.OpenFile(zctx, path, zio.ReaderOpts{})
+	if err != nil {
+		return err
+	}
+	defer zr.Close()
+	wc, err := fs.NewFileReplacer(filepath.Join(p.logdir, "eve.zng"), os.FileMode(0666))
+	if err != nil {
+		return err
+	}
+	defer wc.Close()
+	zw := zngio.NewWriter(wc, zngio.WriterOpts{})
+	if err := zbuf.CopyWithContext(ctx, zw, zr); err != nil {
+		return err
+	}
+	return zw.Close()
```

Comment:
Oops, I should have remembered last night that there's a wrapper which will handle all of the file closes internally:
```suggestion
	return fs.ReplaceFile(filepath.Join(p.logdir, "eve.zng"), os.FileMode(0666), func(w io.Writer) error {
		zw := zngio.NewWriter(zio.NopCloser(w), zngio.WriterOpts{})
		return zbuf.CopyWithContext(ctx, zw, zr)
	})
```